src/service/chat.js

```diff
@@ -161,11 +161,55 @@ const getOrCreateAnonymousChannel = async (userId) => {
   }
 };
 
+const moveChatToSigned = async ({oldChannelId, targetUserId, source}) => {
+  try {
+    const response = await api.post('/chat/move-to-sign', {oldChannelId, targetUserId, source});
+    if (response.status === 200) {
+      return Promise.resolve(response.data);
+    }
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in the `moveChatToSigned` function could be improved. Currently, it only checks for an error message in `e?.response?.data?.message`. However, there might be cases where the error does not have a response or data property, leading to unhandled exceptions. It would be better to add a generic error message as a fallback.
```diff
-    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
-    return Promise.reject(e);
+    const errorMessage = e?.response?.data?.message || 'An error occurred while moving chat to signed';
+    return Promise.reject(errorMessage);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->